system-design/app-interoperability/application-package-definition.md

```diff
@@ -17,98 +28,186 @@ The application package has the following folder structure:
+**Application Metadata Attributes**
 
-| Attribute       | Type            | Required?       | Description     |
-|-----------------|-----------------|-----------------|-----------------|
-| repoType        | string          | Y               | The type of repository where the deployment definition can be found. Options are: http, git, local. |
-| url             | string          | Y               | The URL to the deployment definition. In case of 'docker-compose', the URL points at the `compose.yaml` file.|
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| descriptionFile  | string          | N               | Link to the file containing the application's full description. The file should be a markdown file.|
+| icon             | string          | N               | Link to the icon file (e.g., in PNG format).|
+| licenseFile      | string          | N               | Link to the file that details the application's license. The file should either be a plain text, markdown or PDF file.|
+| releaseNotes     | string          | N               | Statement about the changes for this application's release. The file should either be a markdown or PDF file.|
+| site             | string          | N               | Link to the application's website.|
+| tagline          | string          | N               | The application's slogan.|
+
+**Author Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | N               | The name of the application's creator.|
+| email            | string          | N               | Email address of the application's creator.|
+
+**Organization Metadata Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| name             | string          | Y               | Organization responsible for the application's development and distribution.|
+| site             | string          | N               | Link to the organization's website.|
+
+**Component Attributes**
+
+| Attribute        | Type            | Required?       | Description     |
+|------------------|-----------------|-----------------|-----------------|
+| cluster          | []ComponentType    | N               | Component type element for indicating the component for deploying the application on a cluster using Helm. See the [Component Type](#component-type-attributes) section below. |
+| standalone      | []ComponentType    | N               | Component type element for indicating the component for deploying the application on a standalone device using Docker compose. See the [Component Type](#compnent-type-attributes) section below. |
+
+**Component Type Attributes**
```

Comment:
It seems that we define `ComponentType` unnecessarily in a polymorphic fashion. Because, depending on the value of `ComponentType.type` (either `helm.v3` or `docker-compose`), the specification of `ComponentType.properties` is different. However, the parent attributes `cluster` and `standalone` always contain either helm specifications **or** docker-compose specifications. It is never mixed. It would be cleaner to let `cluster` be of the type  `[]ComponentTypeHelmV3` and  `standalone` of the type `[]ComponentTypeDockerCompose`. Both types would omit the attribute `type`.

Reply:
@g0zilla @arne-broering Maybe this is something we can talk about again during Wednesday's call.

Reply:
@g0zilla @arne-broering After re-reading this comment I realized I misunderstood it originally but I would still like to talk about this because I don't like the use of "cluster" or "standalone" as a grouping.

Reply:
We agreed in the telco on this approach.

Reply:
This change has been applied

Reply:
Unfortunately I couldn't participate at the telco. The specification is fine, but I want to note, that we introduce polymorphism here. This is especially relevant when you generate source code and have to compose an OpenAPI specification accordingly. This is perfectly supported there, but it is more complex as having a static class hierarchy. You need a class mapper which maps to the class type depending on the value in `type`. Unfortunately this is solved completely different across the zoo of programming languages. I wouldn't say, we shouldn't do it like this, but I want to put this aspect into considerations.

Reply:
@g0zilla Good point. We can always come back and refine this as we start working on prototypes, blueprints, and the reference implementation if we encounter issues with this. 

In my mind, we should make it a goal to be able to add new deployment types without having to create a new version of this document. We know these are coming so if we can position ourselves in a way that allows this document to evolve seamlessly when we need to support a new deployment type then it puts us in a better position. I'm sure there are other ways of accomplishing this though.